api/src/controllers/comment.ts

```diff
@@ -0,0 +1,143 @@
+import CommentService from "../services/commentService";
+import { Response, Request } from "express";
+import { ValidationUtils } from "../utils/validationUtils";
+import { MyTVListError } from "../errors/MyTVListError";
+
+export class CommentController {
+    private commentService: CommentService;
+
+    public constructor(commentService: CommentService) {
+        this.commentService = commentService;
+    }
+
+    public async listComments(req: Request, res: Response) {
+        const mediaId = req.params["mediaId"];
+        if (!this.validateMediaId(mediaId, res)) return res;
+
+        try {
+            const comments = await this.commentService.listComments(
+                parseInt(mediaId)
+            );
+            return res.status(200).json(comments);
+        } catch (err: any) {
+            console.error(err.message);
+            return this.handleError(err, res);
+        }
+    }
+
+    public async createComment(req: Request, res: Response) {
+        const { content, mediaId, parentId } = req.body;
+        if (!this.validateMediaId(mediaId, res)) return res;
+        if (!this.validateContent(content, res)) return res;
+        if (!this.validateParentId(parentId, res)) return res;
+        if (!ValidationUtils.validateUserLoggedIn(req, res)) return res;
```

Comment:
É necessário varificar isso, se a rota tem o middleware to `auth`?

Reply:
Não perdemos nada por isso. Não vai acontecer de não vir autenticado, mas caso alguma coisa introduza um bug no router ou no middleware, a gente garante que o controller vai tratar o erro apropriadamente